data_warehouse/services/aggregate_data_service.py

```diff
@@ -380,3 +385,40 @@ def table_training_feedback(cls, context):
             )
         file_path = TableExportService.export_training_feedback(data)
         return file_path, '培训反馈表.xls'
+
+    @classmethod
+    def table_training_record(cls, context):
+        '''个人培训记录导出'''
+        request = context.get('request')
+        event_name = request.query_params.get('event_name')
+        event_location = request.query_params.get('event_location')
+        start_time = request.query_params.get('start_time')
+        end_time = request.query_params.get('end_time')
+        matched_records = TrainingRecordService.get_records(
+            request.user, event_name, event_location, start_time, end_time)
+        # prepare data to be written in excel.
+        data = []
+        for record in matched_records:
+            data.append(
+                {
+                    'event_name':
+                    record.campus_event.name
```

Comment:
Every time you access `record.campus_event` or other foreign keys, Django will perform a SQL query, this will lead to a disaster, consider using `.select_related()` to perform sql join so all data will be retrieved in only one query. e.g.:
```
matched_records = TrainingRecordService.get_records(
    request.user, event_name, event_location, start_time, end_time)
matched_records = matched_records.select_related(
    'campus_event', 'off_campus_event', 'event_coefficient`)
```